src/audio/kpb.c

```diff
@@ -628,7 +629,17 @@ static int kpb_reset(struct comp_dev *dev)
 		ret = comp_set_state(dev, COMP_TRIGGER_RESET);
 		break;
 	default:
+		kpb->kpb_no_of_clients = 0;
 		kpb->hd.buffered = 0;
+		kpb->sel_sink = NULL;
+		kpb->host_sink = NULL;
+		kpb->host_buffer_size = 0;
+		kpb->host_period_size = 0;
+
+		for (i = 0; i < KPB_MAX_NO_OF_CLIENTS; i++) {
+			kpb->clients[i].state = KPB_CLIENT_UNREGISTERED;
+			kpb->clients[i].r_ptr = NULL;
```

Comment:
anything to free here ?

Reply:
If I didn't miss anything, `clients[i].r_ptr` is never assigned to a non-null value.
For `kpb->sel_sink` and `kpb->host_sink`, they point to the allocated `comp_buffer` in `dev->bsink_list`.
They don't need to be freed here.